core/iwasm/libraries/lib-wasi-threads/test/stress_test_threads_creation.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2023 Amazon.com Inc. or its affiliates. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <assert.h>
+#include <errno.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <stdbool.h>
+
+enum CONSTANTS {
+    NUM_ITER = 90000,
+    NUM_RETRY = 5,
+    MAX_NUM_THREADS = 8,
+    RETRY_SLEEP_TIME_US = 4000,
+};
+
+unsigned int threads_executed = 0;
+unsigned int threads_creation_tried = 0;
+
+pthread_mutex_t mutex;
+
+bool is_finished[MAX_NUM_THREADS];
+
+void *
+thread_func(void *arg)
+{
+    __atomic_fetch_add(&threads_executed, 1, __ATOMIC_RELAXED);
+
+    pthread_mutex_lock(&mutex);
+    int num = *(int *)(arg);
```

Comment:
won't that line be removed by compiler's optimizer?

Reply:
I don't think so

I mean, num var is used in the next line so you can't remove it's initialising, if I got your question right

Reply:
I don't think so

I mean, num var is used in the next line so you can't remove it's initialising, if I got your question right

Reply:
Ah yes, sorry I didn't provide a full context in this comment. I also assumed we don't need the is_finished array